tour_model_eval/v-measurel_cutoff_bins_single_user.ipynb

```diff
@@ -0,0 +1,271 @@
+   "id": "infrared-accordance",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "len(non_empty_trips), non_empty_trips"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "generic-glasgow",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bin_trips, bins = pipeline.remove_noise(non_empty_trips, radius)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "thick-commissioner",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "logging.debug('The list of bins is %s' % bins)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "macro-round",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# show all user labels in all bins\n",
+    "for bin in bins:\n",
+    "    bin_user_input = (non_empty_trips[i].data[\"user_input\"] for i in bin)\n",
+    "    bin_df = pd.DataFrame(data = bin_user_input)\n",
+    "    print(bin_df)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "tired-athens",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# turn all user_input into list without binning\n",
+    "bin_trips_user_input_ls = pd.DataFrame(data=[bin_trips[i][\"data\"][\"user_input\"] for i in range(len(bin_trips))]).values.tolist()\n",
+    "bin_trips_user_input_ls"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "veterinary-courage",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for i in bin_trips_user_input_ls:\n",
+    "    if i[2] == \"same_mode\":\n",
+    "        print(i)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "coastal-fiction",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# drop duplicate user_input\n",
+    "no_dup_df=pd.DataFrame(data=[bin_trips[i][\"data\"][\"user_input\"] for i in range(len(bin_trips))]).drop_duplicates()\n",
+    "no_dup_df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "stainless-ancient",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# turn non-duplicate user_input into list\n",
+    "no_dup_list = no_dup_df.values.tolist()\n",
+    "no_dup_list"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "narrative-reducing",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# collect labels_true based on user_input\n",
+    "labels_true =[]\n",
+    "for trip in bin_trips_user_input_ls:\n",
+    "    if trip in no_dup_list:\n",
+    "        labels_true.append(no_dup_list.index(trip))\n",
+    "labels_true"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "framed-sector",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# collect labels_pred based on bins\n",
+    "labels_pred = []\n",
+    "for i in range(len(bins)):\n",
+    "    for trip in bins[i]:\n",
+    "        labels_pred.append(i)\n",
+    "labels_pred"
```

Comment:
I think they are in same order.

You can see the `bin_df`
<img width="719" alt="Screen Shot 2021-02-21 at 11 17 43 AM" src="https://user-images.githubusercontent.com/68228318/108635854-8bf3d180-7436-11eb-9eed-ed91162b6189.png">
That's the trips order after binning. 
It is according to the `non_empty_trips`indices in the bins
<img width="992" alt="image" src="https://user-images.githubusercontent.com/68228318/108635914-e42ad380-7436-11eb-82d2-0579929e3646.png">

The following is the list from where I collect `labels_true`, it is indeed according to the indices in `bin_trips`
<img width="1001" alt="image" src="https://user-images.githubusercontent.com/68228318/108635989-58657700-7437-11eb-8359-108292acad58.png">

I do consider the trip order. I think if `bin_trips_user_input_ls` has the same trip order as trips in `bin_df`, then the way I collect labels should be the same. Although I just show part of the trips, but you can see from the picture, the trips are in same order.

Is that clear?

Reply:
I still don't see this.

`bin_user_input` that you have listed above uses `bins` not `bin_trips`, the same as `labels_pred`.
I think that your main argument is that the labels are the same through spot checking. That is good, but it could also happen by chance and for the first few entries. I certainly did not look at every single entry, and I am not sure I would be very accurate at doing so. This is the kind of checking which it is much better for computers to do programatically.

I would feel a lot more confident by comparing the trip indices in the original list directly. Add an index to the original list (if it doesn't have one) and use it as the dataframe index for both `labels_pred` and `labels_true` and verify that the indices are the same.

Reply:
The red list of `bins` above contains the original indices from trips after Naomi filtered out trips that are points. `bin_trips` is a new data list that collects trips according to the indices in bins. 
Which key word is for timestamp that you want me to add?

Reply:
`trip["data"]["start_ts"]` should work as a timestamp. Again, I don't want you to print it, I want you to check it programmatically

Reply:
<img width="784" alt="image" src="https://user-images.githubusercontent.com/68228318/108669747-7ad8ae00-7492-11eb-957b-f6aacfc2f7b1.png">
Is that ok?

Reply:
That looks good, but not using the full power of pandas/numpy. Please add to PR.